cultivation/scripts/flashcore/cli/main.py

```
@@ -52,48 +60,126 @@ def vet(
         )
-    except FlashcardDatabaseError as e:
-        console.print(f"[bold]A database error occurred during ingestion: {e}[/bold]")
-        raise typer.Exit(code=1) from e
+
+    try:
+        _perform_ingestion_logic(re_ingest)
+    except typer.Exit:
+        raise
+    except DatabaseError as e:
+        console.print(f"[bold red]Database Error:[/bold red] {e}")
+        raise typer.Exit(code=1)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)